freud/order.py

```diff
@@ -206,58 +175,54 @@ def compute(self, orientations):
         Args:
             orientations (:math:`\left(N_{particles}, 3 \right)` :class:`numpy.ndarray`):
                 Orientation vectors for which to calculate the order parameter.
-        """   # noqa: E501
+        """  # noqa: E501
         if orientations.shape[1] == 4:
-            raise ValueError('In freud versions >=3.0.0, Nematic.compute() takes '
-                             '3d orientation vectors instead of 4d quaternions.')
-        orientations = freud.util._convert_array(
-            orientations, shape=(None, 3))
-
-        if len(np.where(~orientations.any(axis=1))[0])!=0:
-            warnings.warn('Including zero vector in the orientations array '
-                          'may lead to undefined behavior.',
-                          UserWarning)
-        cdef const float[:, ::1] l_orientations = orientations
-        cdef unsigned int num_particles = l_orientations.shape[0]
-
-        self.thisptr.compute(<vec3[float]*> &l_orientations[0, 0],
-                             num_particles)
+            raise ValueError(
+                "In freud versions >=3.0.0, Nematic.compute() takes "  # noqa: EM101
```

Comment:
```suggestion
                "In freud versions >=3.0.0, Nematic.compute() takes "
```
We should fix the code instead of ignoring EM101: https://docs.astral.sh/ruff/rules/raw-string-in-exception/

Remove the noqa and `ruff check . --fix --unsafe-fixes` should be able to automatically fix this.